src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -883,5 +897,42 @@ private bool IsFailOverable(RequestFailedException rfe)
                    rfe.Status == (int)HttpStatusCode.RequestTimeout ||
                    rfe.Status >= (int)HttpStatusCode.InternalServerError;
         }
+
+        private async Task<Dictionary<string, ConfigurationSetting>> MapConfigurationData(Dictionary<string, ConfigurationSetting> data)
+        {
+            Dictionary<string, ConfigurationSetting> mappedData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (KeyValuePair<string, ConfigurationSetting> kvp in data)
+            {
+                ConfigurationSetting setting = kvp.Value;
+                foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.UserDefinedMappers)
+                {
+                    setting = await func(setting).ConfigureAwait(false);
+                }
+                mappedData[kvp.Key] = setting;
+            }
+
+            return mappedData;
+        }
+
+        private void UpdateWatchedSettings()
+        {
+            _watchedSettings.Clear();
+
+            foreach (KeyValueWatcher changeWatcher in _options.ChangeWatchers)
+            {
+                string watchedKey = changeWatcher.Key;
+                string watchedLabel = changeWatcher.Label;

```

Comment:
>changeWatcher.Label [](http://example.com/codeflow?start=38&length=19)

We need to call `NormalizeNull()` on the label so that it can be compared properly.